onnxruntime/core/providers/rocm/tunable/gemm_hipblaslt.h

```diff
@@ -22,198 +26,241 @@ using onnxruntime::contrib::rocm::blas::GemmFastGeluParams;
+        HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+            mat_b, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &batch, sizeof(batch)));
+        HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+            mat_b, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &stride_b, sizeof(stride_b)));
+        HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+            mat_c, HIPBLASLT_MATRIX_LAYOUT_BATCH_COUNT, &batch, sizeof(batch)));
+        HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatrixLayoutSetAttribute(
+            mat_c, HIPBLASLT_MATRIX_LAYOUT_STRIDED_BATCH_OFFSET, &stride_c, sizeof(stride_c)));
+      }
+
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+          matmul, HIPBLASLT_MATMUL_DESC_TRANSA, &trans_a, sizeof(int32_t)));
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+          matmul, HIPBLASLT_MATMUL_DESC_TRANSB, &trans_b, sizeof(int32_t)));
+
+      // Deduce enable_bias from params
+      auto d_bias = GetBiasFromParams<T>(params);
+      bool enable_bias = d_bias != nullptr;
+
+      hipblasLtEpilogue_t epilogue;
+      switch (activation_type) {
+        case ActivationType::NONE:
+          epilogue = enable_bias ? HIPBLASLT_EPILOGUE_BIAS : HIPBLASLT_EPILOGUE_DEFAULT;
+          break;
+        case ActivationType::RELU:
+          epilogue = enable_bias ? HIPBLASLT_EPILOGUE_RELU_BIAS : HIPBLASLT_EPILOGUE_RELU;
+          break;
+        case ActivationType::GELU:
+          epilogue = enable_bias ? HIPBLASLT_EPILOGUE_GELU_BIAS : HIPBLASLT_EPILOGUE_GELU;
+          break;
+        default:
+          throw std::runtime_error("Unsupported activation type for HipBlasLtMatMul");
+      }
+      HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+          matmul, HIPBLASLT_MATMUL_DESC_EPILOGUE, &epilogue, sizeof(epilogue)));
+
+      if (enable_bias) {
+        HIPBLASLT_RETURN_IF_ERROR(hipblasLtMatmulDescSetAttribute(
+            matmul, HIPBLASLT_MATMUL_DESC_BIAS_POINTER, &d_bias, sizeof(void*)));
+      }
+
+      size_t workspace_size = 0;
+      hipblasLtMatmulAlgo_t algo_i = algo;
+      auto status = hipblaslt_ext::matmulIsAlgoSupported(op_handle,
+                                                         matmul,
+                                                         &alpha,
+                                                         mat_a,
+                                                         mat_b,
+                                                         &beta,
+                                                         mat_c,
+                                                         mat_c,
+                                                         algo_i,
+                                                         workspace_size);
+
+      TUNABLE_OP_RETURN_UNSUPPORTED_ARGUMENT_IF(
+          status != HIPBLAS_STATUS_SUCCESS, "hipBLASLt find_all: algo not supported, index ", std::to_string(i));
```

Comment:
@abudup I believe this is because we are now using a fixed max workspace size. There will be some cases:

> HIPBLAS_STATUS_INVALID_VALUE – If the parameters are unexpectedly NULL, in conflict or in an impossible configuration. **For example, when workspaceSizeInBytes is less than workspace required by the configured algo.**

Reply:
The collection of all algos is deduced from information about data type and transa/transb, which should be a complete set. But only some of them are applicable to each concrete input shape of M/N/K.